felix/dataplane/linux/int_dataplane.go

```diff
@@ -1035,7 +1040,12 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 				config.MaxIPSetSize))
 			dp.RegisterManager(newPolicyManager(rawTableV6, mangleTableV6, filterTableV6, ruleRenderer, 6, config.RulesConfig.NFTables))
 		} else {
-			dp.RegisterManager(newRawEgressPolicyManager(rawTableV6, ruleRenderer, 6, ipSetsV6.SetFilter, config.RulesConfig.NFTables))
+			mgr := newRawEgressPolicyManager(rawTableV6, ruleRenderer, 6, ipSetsV6.MarkDirty, config.RulesConfig.NFTables)
+			ipSetsV6.SetFilter(func(ipSetName string) bool {
+				neededIPSets := mgr.GetNeededIPSets()
```

Comment:
This is a very inefficient way to do this check.  Every call to the filter will build the full "needed sets" `Set` and then throw it away again.  That is `O(n^2)` in the number of IP sets, which is not acceptable.

However, now you've switched to a "pull" model for this, you should be able to do a lot better.  What if you made the API `mgr.IPSetNeeded(name)`?  Then you could do a trivial map lookup inside the manager instead of building the unneeded set.